src/azureFunction/azureFunctionProjectService.ts

```diff
@@ -0,0 +1,156 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as os from 'os';
+import * as vscode from 'vscode';
+import * as af from '../../typings/vscode-azurefunctions.api';
+import * as fs from 'fs';
+import * as mssql from 'vscode-mssql';
+import { azureFunctionsExtensionName, defaultBindingResult, defaultSqlBindingTextLines, genericCollectionImport, sqlBindingResult, sqlConnectionString } from '../constants/constants';
+import * as LocalizedConstants from '../constants/localizedConstants';
+import { AzureFunctionsService } from '../services/azureFunctionsService';
+import * as azureFunctionUtils from '../azureFunction/azureFunctionUtils';
+import * as constants from '../constants/constants';
+import * as path from 'path';
+
+
+// const sqlBindingNugetSource = 'https://www.myget.org/F/azure-appservice/api/v3/index.json';
+// const sqlBindingPackageName = 'Microsoft.Azure.WebJobs.Extensions.Sql';
+const sqlConnectionStringPropertyName = 'SqlConnectionString';
+
+export class AzureFunctionProjectService {
+
+	constructor(private azureFunctionsService: AzureFunctionsService) {
+	}
+
+	public async createAzureFunction(connectionString: string, schema: string, table: string): Promise<void> {
+		const azureFunctionApi = await this.getAzureFunctionsExtensionApi();
+		if (!azureFunctionApi) {
+			return;
+		}
+		if (!await this.isAzureFunctionProjectOpen()) {
+			vscode.window.showErrorMessage(LocalizedConstants.azureFunctionsProjectMustBeOpened);
+			return;
+		}
+
+		// because of an AF extension API issue, we have to get the newly created file by adding
+		// a watcher: https://github.com/microsoft/vscode-azurefunctions/issues/2908
+		const newFilePromise = this.getNewFunctionFile();
+
+		// get function name from user
+		const functionName = await vscode.window.showInputBox({
+			title: 'Function Name',
+			value: 'HttpTrigger1'
```

Comment:
Seems reasonable to have this default to the table name

Reply:
Following how the Create Azure Function via Azure Functions is set up. HttpTrigger# is typically set up as the default value. @dzsquared for thoughts.

Reply:
That would make sense if this was an empty HttpTrigger function, but we're specifically making a function with a SQL binding to a table so leaving it as the default name is unlikely to be what the user wants.